Webapp/sources/blueprints/polymer_novel_compound/routes.py

```diff
@@ -0,0 +1,44 @@
+from flask import Response, jsonify, request
+from flask_login import login_required
+from sources import services
+from sources.auxiliary import abort_if_user_not_in_workbook
+
+from . import polymer_novel_compound_bp
+
+
+@polymer_novel_compound_bp.route("/_polymer_novel_compound", methods=["GET", "POST"])
+@login_required
+def polymer_novel_compound() -> Response:
+    """
+    Adds a novel compound to the database and links it to the workbook of the current reaction.
+    The novel compound data is saved from the data the user enters in the form.
+    This function is triggered when a novel structure is drawn in the sketcher.
+
+    Returns:
+        Flask Response with feedback about the operation.
+    """
+    # get the active workbook and verify user belongs
+    workgroup_name = str(request.form["workgroup"])
+    workbook_name = str(request.form["workbook"])
```

Comment:
could try using request.form.get("workgroup"), this will return None is there is a KeyError where as request.form["workgroup"] will error out